bin/ch-test

```diff
@@ -186,18 +337,35 @@ fi
 
 # Ensure BATS_TMPDIR is set to /tmp (issue #278).
 if [[ -n $BATS_TMPDIR ]]; then
-    if [[ ! $BATS_TMPDIR == '/tmp/'* ]]; then
+    if [[ "$BATS_TMPDIR" != '/tmp'* ]]; then
         fatal "BATS_TMPDIR: must be /tmp but found $BATS_TMPDIR (issue #278)"
     fi
 fi
 
+# Create a directory in user space to hold auto-generated test artifacts
+tmpdir=${HOME}/.charliecloud/ch-test.tmp.$USER
```

Comment:
(1) Call it something more specific than `tmpdir`.
(2) Generally speaking, I recommend avoiding dot files, as they hide filesystem pollution. Principle: "[explicit is better than implicit](https://inventwithpython.com/blog/2018/08/17/the-zen-of-python-explained/)".